src/engine/NeutralOptional.cpp

```diff
@@ -50,6 +50,12 @@ bool NeutralOptional::knownEmptyResult() { return false; }
 // _____________________________________________________________________________
 bool NeutralOptional::supportsLimit() const { return true; }
 
+// _____________________________________________________________________________
+void NeutralOptional::onLimitChanged(
+    const LimitOffsetClause& limitOffset) const {
+  tree_->applyLimit(limitOffset);
+}
+
```

Comment:
Am I seeing this correctly that doing something fancy is only valid if `supportsLimit()` is true, or what is the mechanism
to not apply in particular the offset twice?

Reply:
(comment in the base clase, and think of a less error-prone interface.